application/controllers/contestant/Scoreboard.php

```diff
@@ -41,6 +42,7 @@ public function index()
 		$this->ui['content']['show_institution_logo'] = $contest['show_institution_logo'];
 		$this->ui['content']['problems'] = $res['problems'];
 		$this->ui['content']['frozen'] = (time() > strtotime($contest['freeze_time'])) && (time() <= strtotime($contest['unfreeze_time']));
+		$this->ui['content']['username'] = $this->session->userdata('username_' . $this->setting->get('sess_id'));
```

Comment:
If I remember correctly, you should be able to get the current username as `$this->identity->get_user_name()`

Reply:
`$this->identity->get_user_name()` retrieves the current user 'name', not the current username of the user. Should i add more method?

Reply:
Oh I see. Yes, please

Reply:
Done